docs/how-to/DataScienceScripts.md

```diff
@@ -0,0 +1,154 @@
+
+## Detailed Design
+
+This section provides a detailed, opinionated approach towards organizing the data science code for model training and evaluation with the goal of easing integration into the Model Factory.  The below are the areas to be considered for the Data Science scripts.
+
+### Data Science Code Component Scripts
+
+The Data Science code must be divided into individual scripts based on the functionality. For example, the NYC Data Science code can be divided into modular scripts:
+    1. `train.py`
+    2. `predict.py`
+    3. `score.py`
+    4. Others as needed.
+
+These component scripts can be run independently or in a sequence. These scripts will be stored in the `src` folder of the DSToolkit for that particular model. For example, the NYC model scripts will be stored in the `src/nyc_taxi` folder as below:
+
+![nyc-folder](../media/nyc-folder.png)
+
+
+### Data Science Code
+
+The Data Science code is written in a modular way such that it can be easily integrated into the Model Factory. The code must be well-documented and should follow the best practices of coding. Having a minimal CI workflow in the data science repo as well as the same Dev Container in the Model Factory creates structure for these engineering best practices.
+
+### Data Science Script Inputs and Outputs
+
+The Data Science scripts must define a series of inputs and outputs. The inputs can be either predefined files or outputs from previous / upstream scripts. The outputs are the results generated by the script which can be used in downstream scripts. Refer to [DSScriptInputsAndOutputs](../how-to/DSScriptInputsAndOutputs.md) for more details on DataScienceInputs and Outputs with examples.
+
+Below is the sample code from `mlops/nyc_taxi/src/mlops_pipeline.py` where the outputs from train component is used in the predict and the output of predict in score components:
+
+    ```python
+    train_model_cmp = gl_pipeline_components[0](
+        dataset_folder=pipeline_job_input,
+    )
+    predict_model_cmp = gl_pipeline_components[1](
+        model_folder=train_model_cmp.outputs.model_folder,
+        dataset_folder=pipeline_job_input,
+    )
+    score_model_cmp = gl_pipeline_components[2](
+        pred_folder=predict_model_cmp.outputs.pred_folder,
+    )
+    ```
+
+### Data Science Script Configuration
+
+The Data Science script must have configuration, which defines the configuration parameter required for the data science scripts. The configuration file should be in yaml format and should be easily configurable. For example, the configuration file for the nyc_taxi model will be stored in the `src/nyc_taxi/common/nyc_config.yaml` folder. And the DS scripts can read the configuration file and use the parameters defined in it.  Below is an example of how the `nyc_taxi_config.yaml` file looks like:
```

Comment:
"Below is an example of how the `nyc_taxi_config.yaml` file looks like"

should be

"Below is an example of the `nyc_taxi_config.yaml` file"

Reply:
Thanks @marshallbentley fixed it